### PR TITLE
cmd/govim: Do not increase version if buffer content didn't change

### PR DESCRIPTION
**Pull request by leitzler**
We currently blindly increase buffer version when vim triggers
BufReadPost or BufNewFile (and a didOpen or didChange request to
gopls). Unloading and loading a buffer that is open also triggers
a BufReadPost, and if the content didn't change we shouldn't
increase version.

This change also skip redefining highlights if the version is
indeed increased since the last known diagnostics could have
changed.

Updates #655

### CODE REVIEW

File: cmd/govim/buffer_events.go [hunk 1]
```diff
@@ -28,8 +28,19 @@ func (v *vimstate) bufReadPost(args ...json.RawMessage) error {
 
 	if cb, ok := v.buffers[nb.Num]; ok {
 		// reload of buffer, e.v. e!
-		cb.SetContents(nb.Contents())
 		cb.Loaded = nb.Loaded
+
+		// If the contents is the same we probably just re-loaded an already open buffer.
```
Comment (myitcv) [inline cmd/govim/buffer_events.go]: "If the contents _are_ the same..."

I don't think it's strictly correct to say "open buffer" either. Would it perhaps be more accurate to say "... probably just re-loaded a currently unloaded buffer"?

**Comment by myitcv** [comment]
FWIW the tests that we do add around this probably belong alongside the diagnostics expectation tests:

https://github.com/govim/govim/tree/4208a9805df15a50ba6e714ca806d425351d50bd/cmd/govim/testdata/scenario_diagnostic_assertions

**Comment by leitzler** [comment]
Agree, raised #667 as a reminder.

File: cmd/govim/buffer_events.go (hunk 1 above)
Comment (leitzler) [reply cmd/govim/buffer_events.go]: Good point, updated!

**Pull request merged**